### PR TITLE
Translate Quint `generate`

### PR DESCRIPTION
**Pull request by konnov**
This PR adds translation of Quint `generate` introduced in https://github.com/informalsystems/quint/pull/1455 into `Apalache!Gen`.

- [x] Tests added for any new code
- [x] [Entries added to `./unreleased/`][changelog format] for any new functionality

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

### CODE REVIEW

**Comment by konnov** [comment]
Hey @bugarela, is there any chance I would be able to get a review? It would be great if I could call Apalache from quint without relying on a fork

**Comment by bugarela** [comment]
@konnov Ah, sorry, I missed this one. So is this why the PR on the Quint side has a failing test?

I saw that PR and thought it was still WIP due to the failing test. I'll take a look at both of them still this week!

**Comment by konnov** [comment]
> @konnov Ah, sorry, I missed this one. So is this why the PR on the Quint side has a failing test?
> 
> I saw that PR and thought it was still WIP due to the failing test. I'll take a look at both of them still this week!

I guess, Quint is failing as it is picking up the older version of Apalache.

File: tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala [hunk 1]
```diff
@@ -666,6 +670,19 @@ class Quint(quintOutput: QuintOutput) {
         tlaDomain <- tlaExpression(domain)
         tlaScope <- tlaExpression(scope)
       } yield tla.exists(tlaName, tlaDomain, tlaScope)
+
+    case (QuintDef.QuintOpDef(_, name, "nondet", QuintApp(id, "generate", Seq(bound, _))), scope) =>
+      val elemType = typeConv.convert(types(id).typ)
+      val boundIntConst = intFromExpr(bound)
+      if (boundIntConst.isEmpty) {
+        throw new QuintIRParseError(s"nondet $name = generate($bound) ... expects an integer constant")
```
Comment (thpani) [inline tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala:678]: Why the `...`?

**Review by thpani — APPROVED** [review]
lgtm

File: tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala (hunk 1 above)
Comment (konnov) [reply tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala:678]: There is a nested expression under `nondet`

**Pull request merged**